scheduler/scheduler.go

```diff
@@ -187,6 +187,12 @@ func (s *scheduler) createTask(sch schedule.Schedule, wfMap *wmap.WorkflowMap, s
 
 	// Bind plugin content type selections in workflow
 	err = wf.BindPluginContentTypes(s.metricManager)
+	if err != nil {
```

Comment:
Looking at the errors that were generated in the tests, and doing some testing, we should be binding the content types after validating the metrics and plugins. We already have a method to check if the metrics and plugins exist for the workflow, and we shouldn't try to bind plugin content types until after verifying all the plugins exist.

We could investigate making them more inclusive, but the content types really only focus on workflows that include a processor with additional processor and/or publisher nodes, correct? If the the processor or publisher plugin doesn't exist, there is no way to bind the plugin content types and check for errors. If a metric doesn't exist, but any processor or publisher plugins in the workflow do exist, then you could try to bind the content types and include any errors that may also exist. 

If the workflow does not include any processing, is there any reason to even call BindPluginContentTypes? Right now, if the workflow is collect -> publish, the metrics returned by collect get encoded in the publishJob.Run before PublishMetrics is called.